File: 003-kids-with-greatest-number-of-candies/Solution.swift
class Solution {
    //// https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/submissions/1139444110/
    func kidsWithCandies(_ candies: [Int], _ extraCandies: Int) -> [Bool] {
        var result: [Bool] = []

        guard let max = candies.max() else {
            return []
        }

        for candy in candies {
            result.append((candy + extraCandies) >= max)
        }

        return result
    }
}

File: 003-kids-with-greatest-number-of-candies/Solution.py
from typing import List

class Solution:
    # // https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/submissions/1139444110/
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        result: list[bool] = []
        maxCandy = max(candies)
        for candy in candies:
            totalCandies = candy + extraCandies
            if totalCandies >= maxCandy:
                result.append(True)
                continue
            result.append(False)
        return result